### PR TITLE
React based plugin

### PR DESCRIPTION
**Pull request by akononenko-sumo**

### CODE REVIEW

File: src/utils/keyedErrors.utils.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { KeyedError } from '../types/metricsApi.types';
+
+export function mapKeyedErrorMessage(keyedError?: KeyedError): string {
+  const type = keyedError?.values?.type ?? 'error';
+
+  if (keyedError) {
```
Comment (pwieckowski-sumo) [inline src/utils/keyedErrors.utils.ts:6]: Inversion would make it easier to read:

```
if (!keyedError) {
  return `Unknown ${type} returned from the API`;
}
.... // rest of IF statements 
```

File: src/components/QueryEditor.tsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+export function QueryEditor({ query, onChange, onRunQuery }: Props) {
+  const onQueryTextChange = (changedQueryText: string) => {
+    onChange({ ...query, queryText: changedQueryText });
+  };
+
+  const { queryText } = query;
+
+  return (
+    <div style={{ display: 'flex', gap: '1rem' }}>
```
Comment (pwieckowski-sumo) [inline src/components/QueryEditor.tsx:17]: Don't we have any possibility to use CSS here (SCSS/LESS/CSS Modules/etc)?

File: README.md [hunk 3]
```diff
@@ -1,189 +1,105 @@
...
+# Grafana data source plugin template
 
-This page describes the sumologic-metrics-grafana-datasource plugin, a datasource plugin for Grafana that can deliver metrics from your Sumo deployment to Grafana. After installing and configuring the plugin, you can query and visualize metrics from Sumo in the Grafana user interface. You initiate a query from the Grafana UI, the search runs on Sumo, and results are returned to Grafana.
+This template is a starting point for building a Data Source Plugin for Grafana.
 
-The Grafana backend will proxy all requests from the browser, and send them on to the data source.
+## What are Grafana data source plugins?
 
-This beta version of sumologic-metrics-grafana-datasource contains most planned features, but is not yet complete.
+Grafana supports a wide range of data sources, including Prometheus, MySQL, and even Datadog. There’s a good chance you can already visualize metrics from the systems you have set up. In some cases, though, you already have an in-house metrics solution that you’d like to add to your Grafana dashboards. Grafana Data Source Plugins enables integrating such solutions with Grafana.
```
Comment (pwieckowski-sumo) [inline README.md:7]: You have a typo - should be "and even Sumo Logic"

**Review by pwieckowski-sumo — APPROVED** [review]

File: src/components/QueryEditor.tsx (hunk 2 above)
Comment (akononenko-sumo) [reply src/components/QueryEditor.tsx:17]: we don't have any CSS here, so I think it simplier to keep it this way

File: README.md (hunk 3 above)
Comment (akononenko-sumo) [reply README.md:7]: it's generated :)

File: src/utils/keyedErrors.utils.ts (hunk 1 above)
Comment (akononenko-sumo) [reply src/utils/keyedErrors.utils.ts]: addressed

**Pull request merged**